app/Libraries/OsuAuthorize.php

```diff
@@ -916,28 +918,54 @@ public function checkForumTopicWatch($user, $topic)
         return 'ok';
     }
 
-    public function checkForumTopicCoverEdit($user, $cover)
+    /**
+     * @param  User $user
+     * @param  Topic|TopicCover $object
+     * @return string
+     */
+    public function checkForumTopicCoverEdit($user, $object)
```

Comment:
authentication cache key needs to be updated to handle multiple types of object.